browser/views.py

```diff
@@ -684,24 +707,49 @@ def edit_investigator_record(request,person_id,research_id):
 
     # Retreive investigator data already stored in database and display to the user
     form = InvestigatorForm(initial={'subject': research.subject, 'role': research.role,
-                                             'year': research.year})
+                                             'year': research.year, 'institution_search': research.institution,
+                                     'institution_id': research.institution.id})
+
+    context = {
+        'form': form,
+        'person': person,
+        'investigator_data': research,
+        'type': 'edit_investigator'
+    }
 
     if request.method == 'POST':
         form = InvestigatorForm(request.POST, instance=person)
+        context.update({
+            'form': form
+        })
         if form.is_valid():
+            if form.cleaned_data.get('institution_search'):
+                if not form.cleaned_data.get('institution_id'):
+                    # Check if the checkbox is checked
+                    if form.cleaned_data.get('create_institution'):
+                        institution = Institution.objects.create(
+                            name=form.cleaned_data.get('institution_search'),
+                            changed_by=request.user
+                        )
+                        form.cleaned_data["institution_id"] = institution.id
+                    else:
+                        messages.add_message(request, messages.ERROR,
+                                             'The above institute does not exist in the database. '
+                                        'Please check the checkbox below to create information about the institute.')
+                        return render(request, template, context)
+
+            # If no institution is given, then get id of institute with None as name
+            if not form.cleaned_data.get('institution_id') and not form.cleaned_data.get('institution_search'):
+                form.cleaned_data["institution_id"] = \
+                    Institution.objects.get(name=form.cleaned_data.get('institution_search')).id
+
```

Comment:
the same comments apply as for the add function

Reply:
Added condition for checking if institution name is not given